sormas-ui/src/main/java/de/symeda/sormas/ui/importer/SurveyTokenResponsesImporter.java

```diff
@@ -0,0 +1,141 @@
+package de.symeda.sormas.ui.importer;
+
+import java.beans.PropertyDescriptor;
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+
+import de.symeda.sormas.api.survey.SurveyTokenCriteria;
+import de.symeda.sormas.api.survey.SurveyTokenIndexDto;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import de.symeda.sormas.api.EntityDto;
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Validations;
+import de.symeda.sormas.api.importexport.ImportErrorException;
+import de.symeda.sormas.api.importexport.ImportLineResultDto;
+import de.symeda.sormas.api.importexport.InvalidColumnException;
+import de.symeda.sormas.api.importexport.ValueSeparator;
+import de.symeda.sormas.api.survey.SurveyDto;
+import de.symeda.sormas.api.survey.SurveyTokenDto;
+import de.symeda.sormas.api.user.UserDto;
+import de.symeda.sormas.api.utils.ValidationRuntimeException;
+
+/**
+ * Data importer that is used to import survey token responses.
+ */
+public class SurveyTokenResponsesImporter extends DataImporter {
+
+	private final SurveyDto survey;
+
+	public SurveyTokenResponsesImporter(File inputFile, UserDto currentUser, SurveyDto survey, ValueSeparator csvSeparator)
+		throws IOException {
+		super(inputFile, false, currentUser, csvSeparator);
+		this.survey = survey;
+	}
+
+	@Override
+	protected ImportLineResult importDataFromCsvLine(
+		String[] values,
+		String[] entityClasses,
+		String[] entityProperties,
+		String[][] entityPropertyPaths,
+		boolean firstLine)
+		throws IOException, InvalidColumnException {
+
+		// Check whether the new line has the same length as the header line
+		if (values.length > entityProperties.length) {
+			writeImportError(values, I18nProperties.getValidationError(Validations.importLineTooLong));
+			return ImportLineResult.ERROR;
+		}
+
+		int index = ArrayUtils.indexOf(entityProperties, SurveyTokenDto.TOKEN);
+		if(index == -1){
+			writeImportError(values, I18nProperties.getValidationError(Validations.tokenColumnWasNotFound));
+			return ImportLineResult.SKIPPED;
+		}
+
+		EntityDto newEntityDto;
+
+		SurveyTokenCriteria criteria = new SurveyTokenCriteria();
+		criteria.setSurvey(survey.toReference());
+		criteria.setToken(values[index]);
+		List<SurveyTokenIndexDto> list = FacadeProvider.getSurveyTokenFacade().getIndexList(criteria, 1, 1, null);
```

Comment:
Please introduce a getByToken facade method instead of requesting the same thing twice